### PR TITLE
Move deb package building to before binaries are stripped

### PR DESCRIPTION
**Pull request by umanwizard**

### CODE REVIEW

File: ci/test/build.sh [hunk 1]
```diff
@@ -45,6 +45,36 @@ docker_run "cargo build --locked --release"
...
+    ci_collapsed_heading "Building .deb package"
+    # shellcheck disable=SC2016
+    # TODO (brennan) - fix this version number
+    docker_run 'cargo-deb --no-strip --deb-version 0.1.0-$(git rev-list HEAD | wc -l)-$(git rev-parse HEAD) -p materialized -o target/debian/materialized.deb'
+    aws s3 cp \
+        --acl=public-read \
+        target/debian/materialized.deb \
+        s3://downloads.mtrlz.dev/materialized-"$BUILDKITE_BUILD_NUMBER"-x86_64.deb
+    #TODO (brennan) - remove the s3 upload once we move off Gemfury
+    # and fully onto Bintray.
```
Comment (benesch) [inline ci/test/build.sh:59]: We're fully on Bintray now, right?

File: ci/test/build.sh [hunk 2]
```diff
@@ -45,6 +45,36 @@ docker_run "cargo build --locked --release"
 ci_collapsed_heading "Building test binaries"
 docker_run "cargo test --locked --no-run && cargo test --locked --no-run --message-format=json > test-binaries.json"
 
+if true; then # XXX(brennan) replace this with the line below before landing, it is just here for testing my PR
+#if [[ "$BUILDKITE_BRANCH" = master ]]; then
+    ci_collapsed_heading "Building .deb package"
+    # shellcheck disable=SC2016
+    # TODO (brennan) - fix this version number
```
Comment (benesch) [inline ci/test/build.sh:52]: ```(cargo metadata --format-version=1 --no-deps | jq -r '.packages[] | select(.name == "materialized") | .version | match("[0-9]\\.[0-9]\\.[0-9]") | .string')```

File: ci/test/build.sh [hunk 3]
```diff
@@ -45,6 +45,36 @@ docker_run "cargo build --locked --release"
...
+    docker_run 'cargo-deb --no-strip --deb-version 0.1.0-$(git rev-list HEAD | wc -l)-$(git rev-parse HEAD) -p materialized -o target/debian/materialized.deb'
```
Comment (benesch) [inline ci/test/build.sh:53]: You extract a `COMMIT_INDEX` and `COMMIT_HASH` below; probably should use those here, too.

By the way, convention for local variables is lowercase: `commit_index` and `commit_hash`. Only environment variables get SCREAMING CASE.

File: ci/test/build.sh [hunk 4]
```diff
@@ -45,6 +45,30 @@ docker_run "cargo build --locked --release"
 ci_collapsed_heading "Building test binaries"
 docker_run "cargo test --locked --no-run && cargo test --locked --no-run --message-format=json > test-binaries.json"
 
+if true; then # XXX(brennan) replace this with the line below before landing, it is just here for testing my PR
+#if [[ "$BUILDKITE_BRANCH" = master ]]; then
+    ci_collapsed_heading "Building .deb package"
+    release_version=(cargo metadata --format-version=1 --no-deps | jq -r '.packages[] | select(.name == "materialized") | .version | match("[0-9]\\.[0-9]\\.[0-9]") | .string')
```
Comment (benesch) [inline ci/test/build.sh:51]: s/`release_version=(`/`release_version=$(`/

Comment (umanwizard) [reply ci/test/build.sh:51]: yes, I know, and also line 54 needs double quotes

**Pull request merged**